File: samples/user-service/generated-py/user_service.py
from __future__ import annotations

from pydantic import BaseModel, Field
from typing import Optional
from nexusrpc import service, Operation


class GetUserInput(BaseModel):
    user_id: str = Field(..., serialization_alias="userId")
    """User ID for the user."""


class User(BaseModel):
    """A user."""

    email: Optional[str] = None
    """Email for the user."""

    user_id: Optional[str] = Field(None, serialization_alias="userId")
    """User ID for the user."""


class GetUserOutput(BaseModel):
    user: User


class SetUserInput(BaseModel):
    user: User


class SetUserOutput(BaseModel):
    user: User


class DeleteUserInput(BaseModel):
    user_id: str = Field(..., serialization_alias="userId")
    """User ID for the user."""


@service
class UserService:
    """A service for managing users."""

    get_user: Operation[GetUserInput, GetUserOutput] = Operation(name="getUser")
    """Get a user."""

    set_user: Operation[SetUserInput, SetUserOutput] = Operation(name="setUser")
    """Create or update a user."""

    delete_user: Operation[DeleteUserInput, None] = Operation(name="deleteUser")
    """Delete a user."""

File: samples/user-service/generated-ts/user-service.ts
import * as nexus from "nexus-rpc";

/**
 * A service for managing users.
 */
export const userService = nexus.service("UserService", {
  /**
   * Get a user.
   */
  getUser: nexus.operation<GetUserInput, GetUserOutput>(),

  /**
   * Create or update a user.
   */
  setUser: nexus.operation<SetUserInput, SetUserOutput>(),

  /**
   * Delete a user.
   */
  deleteUser: nexus.operation<DeleteUserInput, void>(),
});

export interface GetUserInput {
  /**
   * User ID for the user.
   */
  userId: string;
}

export interface GetUserOutput {
  user: User;
}

/**
 * A user.
 */
export interface User {
  /**
   * Email for the user.
   */
  email?: string;
  /**
   * User ID for the user.
   */
  userId?: string;
}

export interface SetUserInput {
  user: User;
}

export interface SetUserOutput {
  user: User;
}

export interface DeleteUserInput {
  /**
   * User ID for the user.
   */
  userId: string;
}
